ports/stm32/mboot/main.c

```diff
@@ -464,31 +466,37 @@ static int usrbtn_state(void) {
 #endif
 
 static int mboot_flash_mass_erase(void) {
-    // TODO
+    #if defined(STM32F4) || defined(STM32F7) || defined(STM32H7)
```

Comment:
with the above fix for `APPLICATION_FLASH_LENGTH` this #if guard should be able to be removed